File: Core/Entities/ReviewPost.cs
using System.ComponentModel.DataAnnotations;

namespace Core.Entities
{
    public class ReviewPost : BaseEntity
    {
        [MaxLength(250)]
        public string Title { get; set; } = null!;
        public string Description { get; set; } = null!;
        public string? AdditionalInformation { get; set; }
        public byte RatingOverall { get; set; }
        public string Location { get; set; }
        public double Longitude { get; set; }
        public double Latitude { get; set; }
        public bool IsRecommended { get; set; }
        public List<Photo> Photos { get; set; } = [];
        public List<ReviewCategory> Categories { get; set; } = [];
        public List<ReviewReaction> Reactions { get; set; } = [];
        public List<ReviewRatingFeedback> RatingFeedbacks { get; set; } = [];
        public List<ReviewCommentFeedback> CommentFeedbacks { get; set; } = [];
        public List<RatingDetail> RatingDetails { get; set; } = [];
    }
}

File: client/src/app/shared/models/review/review-post.ts
import { BaseModel } from "../base";
import { Photo } from "../photo";
import { ReviewCategoryModel } from "./review-category";
import { ReviewCommentFeedbackModel } from "./review-comment-feedback";
import { ReviewRatingDetailModel } from "./review-rating-detail";
import { ReviewRatingFeedbackModel } from "./review-rating-feedback";
import { ReviewReactionModel } from "./review-reaction";

export interface ReviewPostModel extends BaseModel {
  title: string;
  description: string;
  additionalInformation: string | null;
  ratingOverall: number;
  location: string;
  longitude: number;
  latitude: number;
  isRecommended: boolean;
  photos: Photo[];
  categories: ReviewCategoryModel[];
  reactions: ReviewReactionModel[];
  ratingFeedbacks: ReviewRatingFeedbackModel[];
  commentFeedbacks: ReviewCommentFeedbackModel[];
  ratingDetails: ReviewRatingDetailModel[];
}
